src/mono/mono/metadata/sgen-mono.c

```diff
@@ -2361,6 +2361,7 @@ sgen_client_scan_thread_data (void *start_nursery, void *end_nursery, gboolean p
 
 		aligned_stack_start = get_aligned_stack_start (info);
 		g_assert (info->client_info.suspend_done);
+#ifndef ENABLE_EXPERIMENT_COOPROOTS
```

Comment:
Can this be a global variable, controllable from command line or environment variable, instead of an ifdef?
Why should I have to rebuild an replace the runtime, just to add/remove a few lines?
Can the ifdef merely control the default value of a global variable instead? But still allow both options no matter the configuration? ifdef is over used.